src/services/analyticsTracking.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from 'react';
+import ReactGA from 'react-ga';
+import { addInfo as addUserSnapInfo } from './usersnap';
+import { gaTrackingID } from 'common/injectGlobals';
+import { merge } from 'lodash';
+
+const debug = process.env.NODE_ENV === 'development';
+
+let GAState = {
+    trackingId: gaTrackingID,
+    userId: null,
+    userRoles: null,
+    clientId: null,
+};
+let timingsStorage = window.localStorage;
```

Comment:
Do we actually need localStorage here? If we can get away with a singleton object that's held in memory I'd be happy with that, unless the business requirement entails LS.

Reply:
idea here is that localStorage is persisted between sessions. That way we could possibly track super long timings and use timing value outliers as possible indications for areas of confusion

Reply:
Yeah that's fair, cool cool.